langchain4j-google-ai-gemini/src/main/java/dev/langchain4j/model/googleai/GeminiService.java

```diff
@@ -1,85 +1,94 @@
 package dev.langchain4j.model.googleai;
 
-//import io.reactivex.rxjava3.core.Observable;
-import okhttp3.OkHttpClient;
-import okhttp3.logging.HttpLoggingInterceptor;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
 import org.slf4j.Logger;
-import retrofit2.Call;
-import retrofit2.Retrofit;
-import retrofit2.converter.gson.GsonConverterFactory;
-import retrofit2.http.Body;
-import retrofit2.http.POST;
-import retrofit2.http.Path;
-import retrofit2.http.Header;
-import retrofit2.http.Headers;
 
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
 import java.time.Duration;
-//import retrofit2.http.Streaming;
 
-interface GeminiService {
-    String GEMINI_AI_ENDPOINT = "https://generativelanguage.googleapis.com/v1beta/";
-    String API_KEY_HEADER_NAME = "x-goog-api-key";
-    String USER_AGENT = "User-Agent: LangChain4j";
+class GeminiService {
+    private static final String GEMINI_AI_ENDPOINT = "https://generativelanguage.googleapis.com/v1beta";
+    private static final String API_KEY_HEADER_NAME = "x-goog-api-key";
 
-    static GeminiService getGeminiService(Logger logger, Duration timeout) {
-        Retrofit.Builder retrofitBuilder = new Retrofit.Builder()
-            .baseUrl(GEMINI_AI_ENDPOINT)
-            .addConverterFactory(GsonConverterFactory.create());
+    private final HttpClient httpClient;
+    private final Gson gson;
+    private final Logger logger;
 
-        OkHttpClient.Builder clientBuilder = new OkHttpClient.Builder()
-            .callTimeout(timeout);
+    public GeminiService(Logger logger, Duration timeout) {
+        this.logger = logger;
+        this.gson = new GsonBuilder().setPrettyPrinting().create();
 
-        if (logger != null) {
-            HttpLoggingInterceptor logging = new HttpLoggingInterceptor(logger::debug);
-            logging.redactHeader(API_KEY_HEADER_NAME);
-            logging.setLevel(HttpLoggingInterceptor.Level.BODY);
+        this.httpClient = HttpClient.newBuilder()
+                .connectTimeout(timeout)
+                .build();
+    }
+
+    public GeminiGenerateContentResponse generateContent(String modelName, String apiKey, GeminiGenerateContentRequest request) throws IOException {
```

Comment:
```suggestion
    GeminiGenerateContentResponse generateContent(String modelName, String apiKey, GeminiGenerateContentRequest request) throws IOException {
```
nit: please remove `public` from the members of this class